flow/Trace.cpp

```diff
@@ -652,11 +652,17 @@ void removeTraceRole(std::string role) {
 }
 
 TraceEvent::TraceEvent( const char* type, UID id ) : id(id), type(type), severity(SevInfo), initialized(false), enabled(true) {}
-TraceEvent::TraceEvent( Severity severity, const char* type, UID id ) : id(id), type(type), severity(severity), initialized(false), enabled(true) {}
+TraceEvent::TraceEvent( Severity severity, const char* type, UID id )
+	: id(id), type(type), severity(severity), initialized(false),
+	  enabled(g_network == nullptr || FLOW_KNOBS->MIN_TRACE_SEVERITY <= severity) {}
 TraceEvent::TraceEvent( TraceInterval& interval, UID id ) : id(id), type(interval.type), severity(interval.severity), initialized(false), enabled(true) {
```

Comment:
Do we want to disable this line as well if the interval severity is too low?

Reply:
Good idea, missed that one.